Neos.ContentRepository.Core/Classes/Feature/RebaseableCommands.php

```diff
@@ -67,11 +72,37 @@ public function separateMatchingAndRemainingCommands(
     }
 
     private static function commandMatchesAtLeastOneNode(
-        MatchableWithNodeIdToPublishOrDiscardInterface $command,
-        NodeIdsToPublishOrDiscard $nodeIds,
+        RebasableToOtherWorkspaceInterface $command,
+        NodeAggregateIds $nodeIdsToMatch,
     ): bool {
-        foreach ($nodeIds as $nodeId) {
-            if ($command->matchesNodeId($nodeId)) {
+        foreach ($nodeIdsToMatch as $nodeId) {
+            /**
+             * This match must contain all commands which are working with individual nodes, such that they are
+             * filterable whether they are applying their action to a NodeIdToPublish.
+             *
+             * This is needed to publish and discard individual nodes.
+             */
+            $matches = match ($command::class) {
+                CreateNodeAggregateWithNodeAndSerializedProperties::class,
+                DisableNodeAggregate::class,
+                EnableNodeAggregate::class,
+                SetSerializedNodeProperties::class,
+                MoveNodeAggregate::class,
+                RemoveNodeAggregate::class,
+                ChangeNodeAggregateName::class,
+                ChangeNodeAggregateType::class,
+                CreateNodeVariant::class,
+                TagSubtree::class,
+                UntagSubtree::class
+                    => $command->nodeAggregateId->equals($nodeId),
+                CopyNodesRecursively::class => $command->nodeAggregateIdMapping->getNewNodeAggregateId(
+                    $command->nodeTreeToInsert->nodeAggregateId
+                )?->equals($nodeId),
+                SetSerializedNodeReferences::class => $command->sourceNodeAggregateId->equals($nodeId),
+                // todo return false instead to allow change to be still kept?
+                default => throw new \RuntimeException(sprintf('Command %s cannot be matched against a node aggregate id. Partial publish not possible.', $command::class), 1645393655)
```

Comment:
Todo - needs test - there is a bug in partial publish and discard which means that if there is a command on the workspace NOT in this list, no partial operation is allowed. This is a bit to hard. Instead simply keeping that change as remainder makes more sense imo?

Reply:
Solved via https://github.com/neos/neos-development-collection/pull/5385/commits/fa58f0c0f2f59cc41a1864ba6808e126f2677fd5